Aesop/Util/Basic.lean

```diff
@@ -1272,8 +1272,13 @@ partial def getUnassignedGoalMVarDependencies (mvarId : MVarId) :
   return (← go mvarId |>.run {}).snd
   where
     addMVars (e : Expr) : StateRefT (HashSet MVarId) MetaM Unit := do
-      let mvars ← getMVarsNoDelayed e
-      modify (·.insertMany mvars)
+      let mvars ← getMVars e
+      let mut s ← get
+      set ({} : HashSet MVarId) -- Ensure that `s` is not shared.
+      for mvarId in mvars do
+        unless ← isMVarDelayedAssigned mvarId do
```

Comment:
That's just the same logic as before, except with a `for`-loop?

The issue I was talking about is when the delayed assignment contains an unassigned metavariable.  We should visit and return that metavariable as well.

Reply:
Okay, I now recurse into the assignment as well. Does the new version look correct?